### ISSUE TITLE
`nat_open` no longer reports correctly on latest `unstable` (v5.1.0)

### ISSUE THREAD

**Issue by jimmygchen** [sigp/lighthouse#5384]
## Description

The `nat_open` metric exposed via the `/metrics` endpoint and the `lighthouse/ui/health` endpoint doesn't seem to be reporting the status anymore on latest `unstable`. It seems to be related to the change from #5314.

## Version

`unstable` (v5.1.0 release candidate)

## Present Behaviour

- `nat_open` metric not available on `/metrics`.
- `nat_open` always `false` on `lighthouse/ui/health` even when the node is reachable from external nodes.

## Expected Behaviour

`nat_open` metric should be available and should return `true` if node is reachable.

**Comment by AgeManning** [comment]
I think we now report two kinds. One for tcp and one for udp. So we have to update our metrics and this RPC request I think.

**Comment by ackintosh** [comment]
Based on my understanding, before #5314, `nat_open` was determined by [`check_nat`](https://github.com/sigp/lighthouse/blob/50c423ad88aa584edf9aca18e5d1d1ee26b9ad1f/beacon_node/lighthouse_network/src/metrics.rs#L167).  [`check_nat`](https://github.com/sigp/lighthouse/blob/50c423ad88aa584edf9aca18e5d1d1ee26b9ad1f/beacon_node/lighthouse_network/src/metrics.rs#L167) would classify it as open (increment NAT_OPEN counter) if **either** of the conditions below were true.

https://github.com/sigp/lighthouse/blob/50c423ad88aa584edf9aca18e5d1d1ee26b9ad1f/beacon_node/lighthouse_network/src/metrics.rs#L172-L173

Now, `nat_open` is determined by [`observe_nat`](https://github.com/sigp/lighthouse/blob/a89ff100af810d2224f15a24bccf40ca43407df0/common/system_health/src/lib.rs#L202).  [`observe_nat`](https://github.com/sigp/lighthouse/blob/a89ff100af810d2224f15a24bccf40ca43407df0/common/system_health/src/lib.rs#L202) classifies it as open (returns true) only if **both** of the conditions below are true:

https://github.com/sigp/lighthouse/blob/a89ff100af810d2224f15a24bccf40ca43407df0/common/system_health/src/lib.rs#L217

So a quick solution that comes to my mind is to update `observe_nat` to return true if **either** of the conditions are true like below:

```diff
-    discv5_nat && libp2p_nat
+    discv5_nat || libp2p_nat
```

What do you think? cc @jxs

**Comment by jxs** [comment]
thanks for noticing and submitting this Jimmy, and Akihito for the fix! Submitted #5427 5427

**Comment by ackintosh** [comment]
The fix has been released in https://github.com/sigp/lighthouse/releases/tag/v5.1.2 🚀
